tests/driver_dht/main.c

```diff
@@ -15,50 +16,49 @@
  *
  * @author      Ludwig Knüpfer <ludwig.knuepfer@fu-berlin.de>
  * @author      Christian Mehlis <mehlis@inf.fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
  *
  * @}
  */
 
-#ifndef DHT_TYPE
-#error "DHT_TYPE not defined"
-#endif
-
-#ifndef DHT_GPIO
-#error "DHT_GPIO not defined"
-#endif
-
 #include <stdio.h>
 
 #include "xtimer.h"
-
 #include "dht.h"
+#include "dht_params.h"
+
+extern dht_t dht_devs[];
 
 int main(void)
 {
-    dht_t dev;
-
     puts("DHT temperature and humidity sensor test application\n");
 
-    printf("Initializing DHT sensor at GPIO_%ld... ", (long)DHT_GPIO);
-    if (dht_init(&dev, DHT_TYPE, DHT_GPIO) == 0) {
-        puts("[OK]\n");
-    }
-    else {
-        puts("[Failed]");
-        return 1;
-    }
-
-    dht_data_t data;
-    float temp, hum;
+    /* periodically read temp and humidity values */
     while (1) {
+        for (unsigned i = 0; i < DHT_NUMOF; i++) {
+            dht_t *dev = &dht_devs[i];
+            int16_t temp, hum;
+            int16_t dec_temp, dec_hum, int_temp, int_hum;
+
+            if (dht_read(dev, &temp, &hum) == -1) {
```

Comment:
If I see it correctly you depend on the auto initialization here, right? Somehow I'd prefer to initialize the device manually here. Now that I'm writing these line, I'm wondering if we "need" an automatic initialization for this device. What is your motivation?